src/hw4_exceptions/src/main/java/com/company/hw/User.java

```diff
@@ -0,0 +1,180 @@
+        books.add(new Novel("T. Shevchenko", "B", 1865, 5, true));
+        books.add(new Novel("T. Shevchenko", "A", 1845, 5, true));
+        books.add(new Novel("T. Shevchenko", "C", 2019, 5, true));
+
+        books.add(new Comics("A. Chornuy", "Fly", 2019, true));
+        books.add(new Comics("K. Dankovych", "My life", 2017, false));
+        books.add(new Comics("H. Nazarov", "Green door of Earth", 2018, false));
+        books.add(new Comics("L. Poznyak", "Family", 1992, true));
+
+        books.add(new Encyclopedia("O. Bilyk", "Anatomy", 2012, true));
+        books.add(new Encyclopedia("P. Ivanov", "Ukrainian World", 2010, false));
+        books.add(new Encyclopedia("M. Orlyk", "The World Of Us", 2017, false));
+        books.add(new Encyclopedia("A. Kvadrat", "Geometry", 2001, true));
+        books.add(new Encyclopedia("Y. Bachynskyi", "Irredenta", 1920, false));
+        books.add(new Encyclopedia("A. Figure", "Universe of Algebra", 2011, true));
+
+    }
+
+
+
+    //1
+    void getAvailableBooks() {
+        for (Library element : books) {
+            if (element.available) {
+                System.out.println(element);
+            }
+        }
+    }
+
+    //2
+    void getFilteredByAuthor(String author) throws NotAnyBooks{
+        List<Library> tempList = new ArrayList<>();
+        for (Library element : books) {
+            if (element.available && element.getAuthor().contains(author)) {
+                tempList.add(element);
+            }
+        }
+        System.out.println(tempList);
+        if (tempList.isEmpty()) {
+            throw new NotAnyBooks("I throw exception");
+        }
+    }
+
+    //3
+    void sortAlpabetically() {
+        books.sort(new SortByBook());
+        for (Library element : books) {
+            System.out.println(element);
+        }
+    }
+
+    //4
+    void hasEarlierBooks(int year) throws NotAnyBooks{
+        for (Library next : books) {
+            if (year < 1604) {
+                throw new NotAnyBooks("I throw exception");
+            }
+            else if(next.getYear() < year){
+                System.out.println(next);
+            }
+        }
+    }
+
+    //5
+    void checkGenre(String genre) throws NotAnyBooks{
+        List<Library> tempList = new ArrayList<>();
+        for (Library next : books) {
+            if (next.getClass().toString().contains(genre)) {
+                tempList.add(next);
+            }
+        }
+        System.out.println(tempList);
+
+        if (tempList.isEmpty()) {
+            throw new NotAnyBooks("I throw exception");
+        }
+
+    }
+
+    //6
+    boolean hasNotBooksByAuthor(String author) {
+        List<Library> tempList = new ArrayList<>();
+        for (Library next : books) {
+            if (!next.getAuthor().contains(author)) {
+                tempList.add(next);
+            }
+        }
+        return books.size() == tempList.size();
+    }
+
+    //7
+    void filteredByGenre() throws NotAnyBooks {
```

Comment:
throws NotAnyBooks  - let's avoid that & catch Exception within the method

Reply:
Fixed